src/backend-api/Controllers/UpdateController.cs

```diff
@@ -229,6 +227,196 @@ public IActionResult EditProgram([FromBody] PostProgramInputModel input, [FromRo
+                    {
+                        // loop through hardware and depending on what type the hardware is, then add the hardware to the specific table. 
+                        // also in the loop, update the hardware history whether the it was assigned or unassigned  
+                        foreach (var hardware in input.HardwareAssigned)
+                        {
+                            switch (hardware.Type.ToLower())
+                            {
+                                case "monitor":
+                                    var mon = _context.Monitor.Find(hardware.ID);
+                                    mon.EmployeeId = emp.EmployeeId;
+                                    mon.IsAssigned = true;
+                                    UpdateHardwareHistory(true, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                                case "peripheral":
+                                    var periph = _context.Peripheral.Find(hardware.ID);
+                                    periph.EmployeeId = emp.EmployeeId;
+                                    periph.IsAssigned = true;
+                                    UpdateHardwareHistory(true, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                                case "computer":
+                                    var comp = _context.Computer.Find(hardware.ID);
+                                    comp.EmployeeId = emp.EmployeeId;
+                                    comp.IsAssigned = true;
+                                    UpdateHardwareHistory(true, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                                case "server":
+                                    var server = _context.Server.Find(hardware.ID);
+                                    server.EmployeeId = emp.EmployeeId;
+                                    server.IsAssigned = true;
+                                    UpdateHardwareHistory(true, emp.EmployeeId, hardware.ID, hardware.Type);
+                                    _context.SaveChanges();
+                                    break;
+                            }
+                        }
+                    }
+                    // checking if any hardware is to be unassigned
+                    if (input.HardwareUnassigned != null)
+                    {
+                        foreach (var hardware in input.HardwareUnassigned)
+                        {
+                            switch (hardware.Type.ToLower())
+                            {
+                                case "monitor":
+                                    var mon = _context.Monitor.Find(hardware.ID);
```

Comment:
If you feel the urge, this updating unassigned and assigned hardware could be combined into a single method and used in every switch statement.